### PR TITLE
LibJS: Add an optimization to avoid needless arguments object creation

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Parser.cpp [hunk 1]
```diff
@@ -657,7 +657,8 @@ RefPtr<FunctionExpression> Parser::try_parse_arrow_function_expression(bool expe
 
     return create_ast_node<FunctionExpression>(
         { m_state.current_token.filename(), rule_start.position(), position() }, "", move(body),
-        move(parameters), function_length, FunctionKind::Regular, body->in_strict_mode(), true);
+        move(parameters), function_length, FunctionKind::Regular, body->in_strict_mode(),
+        /* might_need_arguments_object */ false, /* is_arrow_function */ true);
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Parser.cpp:661]: Doesn't have to be in this PR, but YES/NO class enums seem perfect for here

**Review by IdanHo — APPROVED** [review]
:yakkiss:

File: Userland/Libraries/LibJS/Parser.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Parser.cpp:661]: I think it'd be useful to have something like `FunctionNode::Flags` instead, then we can put `IsStrict`, `IsArrowFunction`, ... in there all at once.
Definitely meant to be temporary though, yep.

Comment (IdanHo) [reply Userland/Libraries/LibJS/Parser.cpp:661]: Yeah, bit flags would also help with all the repeated TemporaryChance statements

**Review by awesomekling — APPROVED** [review]

**Pull request merged**